compiler/Eta/Iface/IfaceSyn.hs

```diff
@@ -334,7 +334,7 @@ data IfaceIdDetails
 {-
 Note [Versioning of instances]
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-See [http://ghc.haskell.org/trac/ghc/wiki/Commentary/Compiler/RecompilationAvoidance#Instances]
+See [https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/compiler/recompilation-avoidance#Instances]
```

Comment:
My tooling ignored everything after the `#`, so it's possible that it's no longer correct, but that shouldn't be a big deal (if someone wants to make further corrections, they're welcome to do so).